AveLevinBinTree

```diff
@@ -0,0 +1,41 @@
+//637. Average of Levels in Binary Tree
+/**
+ * Definition for a binary tree node.
+ * struct TreeNode {
+ *     int val;
+ *     TreeNode *left;
+ *     TreeNode *right;
+ *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
+ *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
+ *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
+ * };
+ */
+class Solution {
+public:
+    vector<double> averageOfLevels(TreeNode* root) {
+        vector<double> result;
+        queue<TreeNode*> q;    //create a queue to record the node in the same level
```

Comment:
Note: what are the differences between queue and dequeue?

Reply:
Another question: what are the differences between map and unordered_map? (same as set vs. unordered_set)